src/core/server/elasticsearch/version_check/ensure_es_version.ts

```diff
@@ -142,13 +161,19 @@ export const pollEsNodesVersion = ({
       ).pipe(
         map(({ body }) => body),
         catchError((_err) => {
+          if (isUnauthorizedError(_err)) {
+            nodesInfoError = _err as UnauthorizedError;
+          } else {
+            nodesInfoError = _err as Error;
+          }
+
```

Comment:
I'd rather return the error from here instead of changing a variable from a higher scope.

You can see in your last test (https://github.com/TinaHeiligers/kibana/pull/57/files#diff-5fa607800b4c71d8595ccca8677ef365f8bdadb20af935572d0bc21ee096e3a9R225), you had to throw a different error for it to emit after a success.

I'm thinking of a scenario when the connection is not stable and we're getting OK/Connection Error/OK/Connection Error. The connection error is likely to be the same, I don't think we should hide it.